homeassistant/components/geniushub/__init__.py

```diff
@@ -178,7 +178,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: GeniusHubConfigEntry) ->
             password=entry.data[CONF_PASSWORD],
             session=session,
         )
-        unique_id = entry.data[CONF_MAC]
+        assert entry.unique_id is not None
+        unique_id = entry.unique_id
```

Comment:
We could also do `unique_id = entry.unique_id or entry.entry_id` outside the `if..else` and then we can drop the assert too